RackApp/test/test.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+require 'active_support/json'
+require 'nokogiri'
+require 'active_support/core_ext'
+
+OUTER_APP = Rack::Builder.parse_file('config.ru').first
+
+class TestApp < TestBase
+  def app
+    OUTER_APP
+  end
+
+  def test_jsonoutput
+    puts Time.zone = 'UTC'
+    puts @timezone = Time.zone.now
+    data = { 'name' => 'hema', 'time' => @timezone.to_s }
+    json_data = JSON.parse(data.to_json).to_s
+    get '/users/nickname/hema?timezone=UTC&format=json'
+    puts response = JSON.parse(last_response.body).to_s
+    assert_equal response, json_data
+  end
+
+  def test_xmloutput
+    puts Time.zone = 'UTC'
+    puts @timezone = Time.zone.now
+    data = { 'person' => 'hema', 'time' => @timezone.to_s }
+    puts data1 = data.to_xml(root: 'output')
+    get '/users/nickname/hema?timezone=UTC&format=xml'
+    assert_equal last_response.body, data1
+  end
+
+  def test_htmloutput
+    personname = 'hema'
+    Time.zone = 'UTC'
+    puts @timezone = Time.zone.now
+    data = <<-HTML
```

Comment:
Comparing the whole website is not the best idea. You would need to change this test every time you change a view HTML. Better to use https://apidock.com/ruby/Test/Unit/Assertions/assert_match and check for just small strings, something like: 

assert_match "name is #{personname} and current", last_response.body

got the idea?